sympy/solvers/tests/test_solveset.py

```diff
@@ -805,3 +806,101 @@ def test_improve_coverage():
 
     assert _has_rational_power(sin(x)*exp(x) + 1, x) == (False, S.One)
     assert _has_rational_power((sin(x)**2)*(exp(x) + 1)**3, x) == (False, S.One)
+
+
+def test_linear_eq_to_matrix():
+    x, y, z = symbols('x, y, z')
+    eqns1 = [2*x + y - 2*z - 3, x - y - z, x + y + 3*z - 12]
+    eqns2 = [Eq(3*x + 2*y - z, 1), Eq(2*x - 2*y + 4*z, -2), -2*x + y - 2*z]
+
+    A, b = linear_eq_to_matrix(eqns1, x, y, z)
+    assert A == Matrix([[2, 1, -2], [1, -1, -1], [1, 1, 3]])
+    assert b == Matrix([[3], [0], [12]])
+
+    A, b = linear_eq_to_matrix(eqns2, x, y, z)
+    assert A == Matrix([[3, 2, -1], [2, -2, 4], [-2, 1, -2]])
+    assert b == Matrix([[1], [-2], [0]])
+
+
+def test_linsolve():
+    x, y, z, u, v, w = symbols("x, y, z, u, v, w")
+
+    # Test for different input forms
+    x1, x2, x3, x4 = symbols('x1, x2, x3, x4')
+
+    M = Matrix([[1, 2, 1, 1, 7], [1, 2, 2, -1, 12], [2, 4, 0, 6, 4]])
```

Comment:
I would like to see either most of the tests with all symbols instead of integers or at least one nice fully symbolic test.